File: spec/php/OpaqueWithParamTest.php
<?php
namespace Kaitai\Struct\Tests;

class OpaqueWithParamTest extends TestCase {
    public function testOpaqueWithParam() {
        $r = OpaqueWithParam::fromFile(self::SRC_DIR_PATH . "/term_strz.bin");

        $this->assertSame('foo|b', $r->one->buf);
        $this->assertSame(0x61, $r->one->trailer);
    }
}

File: spec/go/spec/opaque_with_param_test.go
package spec

import (
	"runtime/debug"
	"os"
	"testing"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	. "test_formats"
	"github.com/stretchr/testify/assert"
)

func TestOpaqueWithParam(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			debug.PrintStack()
			t.Fatal("unexpected panic:", r)
		}
	}()
	f, err := os.Open("../../src/term_strz.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)
	var r OpaqueWithParam
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, "foo|b", r.One.Buf)
	assert.EqualValues(t, 97, r.One.Trailer)
}
